File: frontend/src/models/notification/UserNotification.ts
export default interface Notification extends UserNotification {
  id: string
}

// this is what we get from the server
export interface UserNotification {
  type: UserNotificationType
  title: string,
  message?: string,
  userId?: string, // the server also always sets this
  groupId?: string,
  groupName?: string,
}

type UserNotificationType = "SUCCESS" | "ERROR"

File: backend/notification-service/src/main/kotlin/edu/locationsharing/models/UserNotification.kt
package edu.locationsharing.models

class UserNotification(
    val type: Type,
    val title: String,
    val message: String? = null,
    val userId: String,
    val groupId: String?= null,
    val groupName: String? = null,
) {

    enum class Type {
        SUCCESS, ERROR
    }

    override fun toString(): String {
        return """
            SystemNotification {
                type: $type,
                title: $title,
                message: $message,
                userId: $userId,
                groupId: $groupId,
                groupName: $groupName
            }
        """.trimIndent()
    }
}
